### PR TITLE
Make the up/down behavior with arrows built in

### PR DESCRIPTION
**Pull request by blink1073**
But also allow additional bindings.

Adds shift+up/down behavior which augments existing selection.

### CODE REVIEW

File: example/src/index.ts [hunk 1]
```diff
@@ -71,18 +71,18 @@ function main(): void {
     sequence: ['Enter'],
     selector: '.jp-DirListing',
     handler: () => {
-      fbWidget.rename();
+      fbWidget.open();
       return true;
     }
   }, {
-    sequence: ['ArrowDown'],
+    sequence: ['Ctrl N'],
```
Comment (sccolbert) [inline example/src/index.ts]: Is this correct?

File: example/src/index.ts [hunk 2]
```diff
@@ -71,18 +71,18 @@ function main(): void {
...
     selector: '.jp-DirListing',
     handler: () => {
       fbWidget.selectNext();
       return true;
     }
   }, {
-    sequence: ['ArrowUp'],
+    sequence: ['Ctrl P'],
```
Comment (sccolbert) [inline example/src/index.ts]: Is this correct?

File: src/listing.ts [hunk 3]
```diff
@@ -406,9 +406,9 @@ class DirListing extends Widget {
   /**
    * Select next item.
    */
-  selectNext(): void {
+  selectNext(keepExisting = false): void {
```
Comment (sccolbert) [inline src/listing.ts:411]: This flag needs some docs. What does it do?

File: src/listing.ts [hunk 4]
```diff
@@ -431,9 +431,9 @@ class DirListing extends Widget {
   /**
    * Select previous item.
    */
-  selectPrevious(): void {
+  selectPrevious(keepExisting = false): void {
```
Comment (sccolbert) [inline src/listing.ts:438]: This flag needs some docs. What does it do?

File: example/src/index.ts (hunk 2 above)
Comment (blink1073) [inline example/src/index.ts]: Yes, this augments the built in shortcuts with the emacs navigation shortcuts for the purposes of the example.

File: src/listing.ts (hunk 4 above)
Comment (blink1073) [inline src/listing.ts:438]: Added param docs.

File: example/src/index.ts (hunk 2 above)
Comment (sccolbert) [inline example/src/index.ts]: Ok. Can you add an inline comment explaining as much?

Comment (blink1073) [inline example/src/index.ts]: Done.

**Comment by sccolbert** [comment]
lgtm

**Pull request merged**